app/helpers/application_helper/button/mass_transform_vm_button.rb

```diff
@@ -0,0 +1,15 @@
+class ApplicationHelper::Button::MassTransformVmButton < ApplicationHelper::Button::Basic
+  def visible?
+    true
+  end
+
+  def disabled?
+    @error_message = _('No suitable target infra provider exists') unless destination_exists?
+    @error_message.present?
+  end
+
+  def destination_exists?
+    # Is there a provider that supports import?
+    !EmsInfra.all.select(&:validate_import_vm).empty?
```

Comment:
please don't fetch all objects into memory if possible.

Reply:
@ohadlevy Can you please explain how it should be written correctly?

Reply:
see https://apidock.com/rails/ActiveRecord/FinderMethods/exists%3F (asuming its AR based).

Reply:
please limit your query only to RHV providers, also - please answer the question, is this making an api call? what happens if the provider is down / slow etc?

Reply:
@ohadlevy I've checked, this code does not seem to call API, it does not block if oVirt API endpoint is not reachable.

Reply:
@martinpovolny would this be a better option?
```ruby
MiqAeServiceManageIQ_Providers_Redhat_InfraManager.all.detect(&:validate_import_vm)
```

Reply:
@ohadlevy Limited the call.